core-backend/common/src/memory.rs

```diff
@@ -290,7 +288,7 @@ impl<Ext: BackendExternalities> MemoryAccessManager<Ext> {
             self.read_into_buf(memory, read.ptr, &mut buff, gas_counter)?;
             buff
         };
-        let decoded = T::decode_all(&mut &buff[..]).map_err(|_| MemoryAccessError::Decode)?;
+        let decoded = T::decode_all(&mut &buff[..]).expect("Can't decode buffer");
```

Comment:
looks like returning an error is more robust

Reply:
It's unreachable anyway

Reply:
reverted